geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/AlterQueryServiceCommandWithSecurityDUnitTest.java

```diff
@@ -0,0 +1,256 @@
+  private final int serversToStart = 2;
+
+  @Rule
+  public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  @Rule
+  public ClusterStartupRule cluster = new ClusterStartupRule();
+
+  @Rule
+  public GfshCommandRule gfsh = new GfshCommandRule();
+
+  private static MemberVM locator;
+  private static List<MemberVM> servers = new ArrayList<>();
+
+  @Before
+  public void setUp() throws Exception {
+    locator = cluster.startLocatorVM(0, l -> l.withSecurityManager(SimpleSecurityManager.class));
+    int locatorPort = locator.getPort();
+    IntStream.range(0, serversToStart).forEach(
+        i -> servers.add(i, cluster.startServerVM(i + 1, s -> s.withConnectionToLocator(locatorPort)
+            .withCredential("clusterManage", "clusterManage"))));
+    gfsh.connectAndVerify(locator);
+  }
+
+  // This test verifies that the cluster starts with the default method authorizer configured, then
+  // changes the currently configured authorizer to each of the out-of-the-box authorizers.
+  @Test
+  public void alterQueryServiceCommandUpdatesMethodAuthorizer() {
+    Class<RestrictedMethodAuthorizer> restrictedAuthorizerClass = RestrictedMethodAuthorizer.class;
+    Class<UnrestrictedMethodAuthorizer> unrestrictedAuthorizerClass =
+        UnrestrictedMethodAuthorizer.class;
+    Class<JavaBeanAccessorMethodAuthorizer> javaAccessorAuthorizerClass =
+        JavaBeanAccessorMethodAuthorizer.class;
+    Class<RegExMethodAuthorizer> regExAuthorizerClass = RegExMethodAuthorizer.class;
+
+    verifyCurrentAuthorizerClass(DEFAULT_AUTHORIZER_CLASS);
+
+    String authorizerName = unrestrictedAuthorizerClass.getName();
+    gfsh.executeAndAssertThat(COMMAND_NAME + " --" + METHOD_AUTHORIZER_NAME + "=" + authorizerName)
+        .statusIsSuccess().hasTableSection().hasRowSize(serversToStart);
+
+    verifyCurrentAuthorizerClass(unrestrictedAuthorizerClass);
+
+    authorizerName = javaAccessorAuthorizerClass.getName();
+    gfsh.executeAndAssertThat(COMMAND_NAME + " --" + METHOD_AUTHORIZER_NAME + "=" + authorizerName
+        + " --" + AUTHORIZER_PARAMETERS + "=" + JAVA_BEAN_AUTHORIZER_PARAMS).statusIsSuccess()
+        .hasTableSection().hasRowSize(serversToStart);
+
+    verifyCurrentAuthorizerClass(javaAccessorAuthorizerClass);
+    servers.forEach(server -> server.invoke(() -> {
+      JavaBeanAccessorMethodAuthorizer methodAuthorizer = (JavaBeanAccessorMethodAuthorizer) Objects
+          .requireNonNull(ClusterStartupRule.getCache()).getService(QueryConfigurationService.class)
+          .getMethodAuthorizer();
+
+      Set<String> expectedParameters =
+          new HashSet<>(Arrays.asList(JAVA_BEAN_AUTHORIZER_PARAMS.split(",")));
+      assertThat(methodAuthorizer.getAllowedPackages()).isEqualTo(expectedParameters);
+    }));
+
+    authorizerName = regExAuthorizerClass.getName();
+    gfsh.executeAndAssertThat(COMMAND_NAME + " --" + METHOD_AUTHORIZER_NAME + "=" + authorizerName
```

Comment:
Maybe break into a separate helper method since we do the same 3-ish steps multiple times in this test.

Reply:
Done!